docs/environment-and-software/vscode.md

```diff
@@ -0,0 +1,197 @@
+
+![Extension Details](vscode/media/extensions-list.png)
+
+!!! info 
+    Extensions are a powerful tool within the VS Code environment but not all extensions are sponsored by reputable sources.  Extension problems can manifest in errors, poor memory management, and excessive CPU utilization.  We are unable to support all possible extensions that exist so we recommend disabling all but the core extensions before reporting a problem to the NCAR HPC support team.
+
+## Conda and Python Environments
+
+### Terminal mode
+
+Terminal mode ignores the VS Code environment and uses the NCAR system module stack.  This is identical to the behavior of running within the Mac Terminal over SSH.  You can perform module loads and activate Conda environments directly from the terminal.
+
+!!! tip 
+    Interacting with a file that was opened with the `code` command will require the Interactive method for running the code.
+
+### Interactive mode
+
+VS Code can use the Python Interactive mode to launch Jupyter notebooks, Python scripts, or debug code with the proper extensions.  
+
+The *Python Interpreter* is used to set the environment for interactive mode.  To select an environment, use the *Python: Select Interpreter* command from the Command Palette (⇧⌘P).  A valid environment is required to properly run Jupyter notebooks within VS Code and is a common use case for NCAR users.
+
+If you are connected to Derecho or Casper and have a Jupyter notebook or Python file preview open, you can click *Select Kernel* to view your available Python environments.  The list includes NCAR conda environments like `npl` and your own custom environments.
+
+![Kernel Select](vscode/media/selectkernel.png)
+
+Open a Jupyter notebook or Python file using either the terminal command `code` or with the File Explorer.  This will put you in the default kernel.  Like Jupyterhub, you will need to specify which kernel to use when executing cells in the notebook.  Not specifying a kernel will use your default kernel and could lead to import failures or unexpected behavior.
+
+!!! info 
+    Adding your conda-envs folder to the list of *Trusted* kernels will allow you to run notebooks with these kernels without being prompted to trust the kernel each time you run select a new kernel.
+
+## Jupyter Notebooks
+You can run cells similarly to Jupyterhub with the 'Run All' or an individual cell by hitting *Shift+Enter*.  Jupyter notebooks will use the VS Code kernel as your Python environment.
+
+![Jupyter Notebook](vscode/media/jnbkernel.png)
+
+Be mindful of the location of launching your notebooks.  If you use the File Explorer tab then you will be running on the login nodes.  
+Executing the command *code* from an interactive job will execute on the compute node.
+
+## Connecting to Compute Nodes
```

Comment:
This looks great. thanks for adding this. My only suggestion is in adding a link to this section in the early stages of the document where there is a message about login nodes memory usage. Something along the line of " If you are interested in using vscode with more memory, please check this section (links to this section).

Reply:
I've added a link in the danger field in the login node section.  I also moved the connect to compute node section to right after connecting to the login nodes as it flows better.

Reply:
That is a great idea. Thanks for updating it. I agree that it flows better.